Improve concurrency safety description of LogRecordProcessor.OnEmit (#4578)
open-telemetry/opentelemetry-specification: specification/logs/sdk.md

For a `LogRecordProcessor` registered directly on SDK `LoggerProvider`,
the `logRecord` mutations MUST be visible in next registered processors.

A `LogRecordProcessor` may freely modify the `logRecord` for the duration of

> the `OnEmit` call. However, `ReadWriteLogRecord` is **not** required to be

Author: WDYT of 
```suggestion
the `OnEmit` call. However, `ReadWriteLogRecord` is MAY NOT be
```

> concurrent-safe. Therefore, any modification to `logRecord` concurrent

Reviewer: This implies that the lack of concurrent-safety is the primary reason for discouraging this, and that the race conditions mentioned in the next sentence are the result of a lack of concurrent-safety. It follows that if a language does guarantee concurrent-safety, that asynchronous modification would be acceptable.

But even if concurrent-safety is guaranteed, we still want to discourage asynchronous modification because it leads to unintuitive and surprising behavior. 

I think we need to rephrase to make this clear.

> with `OnEmit` or after `OnEmit` returns (for example, during asynchronous

Reviewer: What do you mean by "concurrent with `OnEmit`"? Seems like this clause could just be removed.

Author: I totally agree!

processing) may result in race conditions. To avoid such race conditions,

> it is recommended to use a clone of `logRecord` for any asynchronous

Reviewer 2: ```suggestion
implementations SHOULD recommended to users that a clone of `logRecord` be used for any asynchronous
```
